framework/src/constraints/EqualValueBoundaryConstraint.C

```diff
@@ -88,9 +94,60 @@ EqualValueBoundaryConstraint::updateConstrainedNodes()
   _primary_node_vector.clear();
   _connected_nodes.clear();
 
+  // user provided nothing
   if ((_secondary_node_ids.size() == 0) && (_secondary_node_set_id == "NaN"))
     mooseError("Please specify secondary node ids or boundary id.");
-  else if ((_secondary_node_ids.size() == 0) && (_secondary_node_set_id != "NaN"))
+
+  // Find primary node id from user-specified primary node coordinates
+  if (_primary_node_id == std::numeric_limits<unsigned int>::max() &&
+      isParamSetByUser("primary_node_coord"))
+  {
+    const Real eps = libMesh::TOLERANCE;
+
+    std::unordered_set<dof_id_type> local_primary_node_ids;
+    // Gather local candidates
+    for (const auto & bd_node : *_mesh.getBoundaryNodeRange())
+    {
+      if ((*(bd_node->_node) - _primary_node_coord).norm() < eps)
+      {
+        // either the node id should be in the secondary node ids list
+        // or the node should be on the secondary boundary
+        if (std::find(_secondary_node_ids.begin(),
+                      _secondary_node_ids.end(),
+                      bd_node->_node->id()) != _secondary_node_ids.end() ||
+            bd_node->_bnd_id == _mesh.getBoundaryID(_secondary_node_set_id))
+        {
+          local_primary_node_ids.insert(bd_node->_node->id());
```

Comment:
It seems like this conditional is determining whether a boundary node corresponds to the secondary node set, not the primary node set

Reply:
I completed rewrote this part of EqualValueBoundaryConstraint. Many old constructs are outdated, and the logic became unclear as we accept changes over the years.